src/core/mac/mac_frame.cpp

```diff
@@ -972,6 +972,46 @@ const TimeIe *Frame::GetTimeIe(void) const
 }
 #endif // OPENTHREAD_CONFIG_TIME_SYNC_ENABLE
 
+bool Frame::IsAddrMatch(otPanId           aPanId,
+                        otShortAddress    aShortAddress,
+                        const ExtAddress &aExtAddress,
+                        bool              aRxOnWhenIdle) const
+{
+    bool         rval = false;
+    Mac::Address dst;
+    Mac::PanId   panid;
+
+    SuccessOrExit(GetDstAddr(dst));
+
+    // Verify address match
+    switch (dst.GetType())
+    {
+    case Address::kTypeShort:
+        VerifyOrExit((aRxOnWhenIdle && dst.IsBroadcast()) || dst.GetShort() == aShortAddress, OT_NOOP);
```

Comment:
I am not sure about moving this logic in a method in `Mac::Frame` class. I suggest we keep it in `Mac` class (as before).
- `Mac:Frame` class is a representation of a frame.
- Make sense for it to provide methods to parse the frame, and get/set different fields in the frame.
- Doing such checks that are tied to whether `aRxOnWhenIdle` does not fit well here.
- The `aRxOnWhenIdle` is really a `Mac` state variable and `Frame` should not be aware of it or care about it. 

I understand that you moved this this here so that it can be reused from `examples/platform/utils` library code. I think the `platform/util` code is behaving more like MAC layer itself than just frame parser.  In this case, I would argue that having a clean model in what `Mac::Mac` class should do, and what `Mac::Frame` class should provide,  is more valuable over maybe not being able to share a bit of code between platform util  library and OT core.

Reply:
That make sense to me. But the cost is some duplication of code. I would like to get more feedback on this. @bukepo @jwhui , how do you think?

Reply:
Another way of thinking this problem, we can consider this address filter as a `Mac` method and is only needed in `Mac` layer.

For `Radio` which needs to do address filtering, most of the case, it will be done at Chip hardware which does not need this utility function. For simulation radio, as it just simulate a hardware behavior, we can add this filtering(with some code duplication) at simulation radio which is dedicate for simulation radio and **delete it from `Mac:Frame` class**. This way, we can save the code space in `Mac:Frame` class and does not increase the code size for `Mac` layer. How do you think? @abtink

Reply:
This sounds good to me. Thanks @brianljt .